### ISSUE TITLE
Eval: Technical Debt

### ISSUE THREAD

**Issue by mtrythall** [djangocon/2019.djangocon.us#6]
We need to evaluate the current frontend build system:

1. [x] Is gulp still something we want to use? (probably not)
2. [x] Can CSS compilation be easier?
3. [ ] ~~What doc improvements do we need to make to help other devs make contributions?~~
4. [x] Can we reduce our JS dependencies?
5. [x] Updating dependencies to new versions

More as we think of them.

This ticket should deal with the assessment of the system and subsequent tickets should deal with the resolution. Decisions should be outlined here with related ticket.

We will create a Milestone if necessary, but evaluation for now 😄

**Comment by mtrythall** [comment]
Did some digging through the system and here's what I got:

1. Ditch node-sass in favor of Dart Sass, which is a JS-first implementation and no Python 2.7 dependency.
2. Ditch Style Sherpa. The style guide would just be raw HTML. Since it changes so seldom, clean markup + comments would go a long way.
3. Ditch gulp and have all compilation of Sass/other systems handled via CLI. There would be `npm run build-css` or something similar would be all you need to compile everything.

Still thinking through doc improvements, but I think we can simplify just about everything.

**Comment by jefftriplett** [comment]
I hear good things about https://parceljs.org/ which (I know, I know) apparently just works.

**Comment by mtrythall** [comment]
I moved us over to Parcel. I was 50/50 on it until I thought about how it'll simplify JS bundling. Versioning files is a pain.

Parcel has to be installed globally for it to work, so I'm not sure what the impact is on deployment yet. @jefftriplett any thoughts?

As a side note, I'm marking the documentation requirement as Done as I've worked through this enough times and I think it's fine. What I _will_ do is updates the documentation a bit as I'm working through theming, but I can't identify any major issues/areas in discovery. That satisfies this ticket.
